src/main/kotlin/com/devfloor/untitled/article/application/ArticleFacade.kt

```diff
@@ -17,17 +15,14 @@ class ArticleFacade(
     private val articleHashtagService: ArticleHashtagService,
     private val articleFavoriteService: ArticleFavoriteService,
     private val articleOptionService: ArticleOptionService,
-    private val optionService: OptionService,
     private val hashtagService: HashtagService,
 ) {
     @Transactional
     fun create(request: ArticleRequest, user: User): Long {
         val article = articleService.create(request.toArticle(request, user))
-        if (request.isOptionsNotEmpty()) {
-            optionService.showAllByOptionType(request.options)
-                .map { ArticleOption(article, it) }
-                .let { articleOptionService.createAll(it) }
-        }
+        if (request.isOptionsNotEmpty())
+            articleOptionService.createAll(article, request.options)
+
```

Comment:
이 부분은 이전에 로직으로 두는게 개인적으론 더 나은 것 같아.
그리고 if 문의 경우 `{}` 중괄호를 항상 사용하는 건 어떨까? 이게 java 컨벤션에서는 항상 `{}` 중괄호를 사용해라고 하는데, kotlin도 java랑 같은 결을 가지는 언어이다 보니 이런 컨벤션을 지키는 게 좋지 않을까? 싶었어

Reply:
이전에 로직이라면 아래의 것일텐데, 이미 modify때문에 ArticleOptionService에서 만들어놓은 메서드가 있음에도 사용하지 않는게 더 좋다고 생각하는 이유가 뭐야??
```suggestion
        if (request.isOptionsNotEmpty()) {
            optionService.showAllByOptionType(request.options)                
              .map { ArticleOption(article, it) }                
              .let { articleOptionService.createAll(it) }
        }

```

Reply:
kotlin가 java 기반으로 만들어졌긴 하지만, 둘은 명백히 다른 언어인데 그 컨벤션까지 따라가야 할 필요가 있을까?

one line if문에 {}를 빼는게 error-prone인건 맞긴하지만, if문 내부에 더 늘어날 코드는 없지 않을까 생각하고! (만약 있다고 해도, 여기서 IDE가 인덴트까지 잡아주는데 실수를 할 것 같지도 않고)
한줄짜리 if문에서 indent를 통해 이미 구별이 되는데 굳이 {}를 넣는건 오히려 빈 줄 하나 넣는거고 낭비라고 생각해.

그리고, 이러한 코딩 스타일이 컨벤션으로 정해져있지 않은 이상 이건 어디까지나 각자의 스타일이라고 생각하는데,
가독성을 크게 해치지 않는 이상은 스타일에 대한 논의는 최대한 줄이는게 좋지 않을까?

Reply:
첫번째 로직 부분에 대한 의견은 내가 ArticleOptionService 코드를 제대로 안봤었네. 가져다가 써도 충분할 것 같아. 추가적으로 내가 제안하는 방법으로는 현재 `ArticleOptionService`의 `createAll` 메서드가 `List<String>` 타입으로 `options`를 받아오는데 그것보다는 차라리 `List<Option>`을 넘겨주는 식으로 고치면 어떨까?

그럼 해당 코드가
```kotlin
fun createAll(article: Article, options: List<Option>): List<ArticleOption> {
    return options.map { ArticleOption(article, it) }
        .let { repository.saveAll(it) }
```
이렇게 수정될 건데, 그럼 ArticleOptionService에서 OptionService를 가질 필요가 없어서 더 코드가 의존성 없이 관리하기 좋지 않을까?

Reply:
그리고 두 번째 의견에 대해서는

첫 번째로 kotlin이 java를 보완하면서 대체하기 위한 목적으로 나온 언어인데 명백히 다른 언어라고 할 수 있을까? kotlin에서 내새우는 장점이 java 코드와 100% 호환되면서 같이 사용할 수 있다는 부분을 장점으로 내세우는 언어인 만큼 난 충분히 많은 부분이 유사하다고 생각하고 실제로 그러하다고 느끼기도 하고.

그리고 두 번째로 스스로 error-prone인 걸 알고 있다고 했는데 단순한 코드 스타일의 문제라고 생각하는 부분이 조금 의아하긴 했어.

그리고 세 번째는 아마 내가 우테코에서 배운 교육의 가장 첫 시작이자 기본이 컨벤션을 지키는 코딩을 하는 것으로 배워서 그럴 수도 있어. 위에 내가 말한 if에서 한 줄이라도 `{}`을 사용하자는 건 google java coding convention에 명시되어 있는 내용이었거든. 물론 그냥 스타일로 치부할 수도 있지만 우리가 교육 시작하고 현직자로부터 코드 리뷰를 받았을 때 이런 걸 코드 리뷰에서도 지적 받았거든. 내가 생각했을 때 그 이유는 이런 사소한 부분 하나하나를 지키면서 코드를 작성하는 것이 나중에 발생할 에러를 사전에 방지할 수 있는 더 좋은 방법이라고 다들 알고 그래서 지키기 떄문이라서 그러지 않았을까 싶었어.

그리고 내가 코틀린 언어를 공부할 때 if문에서 `{}`을 제거할 수 있도록 지원한 이유는 if가 더 이상 **문**이 아니라 **식**이기 때문에 이러한 방법을 지원해줬다고 알고 있어. java에서는 if는 반환값을 가질 수 없는 문이었는데 kotlin에서는 if는 반환값을 가지는 식으로 사용할 수 있고 그런 부분에서 `{}`을 강제로 가지게 하기엔 코드 가독성에 어색함을 가질 수 있을 것 같아서 제거를 허용한 걸로 공부했었어. 뭔가 그런 관점에서 보다 보니 더 어색함을 많이 느껴서 내가 이렇게 리뷰를 남겼던 것 같아.

물론 개인적인 코드 스타일로 볼 수도 있겠지만 그래도 컨벤션으로 자리잡고 있는 부분들은 많은 사람들이 그렇게 사용하고 있고 이런 식으로의 코딩이 익숙할 수 있다는 말이 될 수도 있으니까 한 번 참고해보면 좋을 것 같아.

Reply:
if에서 `{}`을 사용하는 건 너무 내가 당연히 지켜야 한다고 배우고 알고 있던 부분이라 사실 내 생각이 맹신일 수도 있을 것 같다는 생각이 들긴 하네. 근데 뭔가 아래 두 문서 예시를 참고했을 때도 내 의견이 괜찮은 의견이라고 생각이 들긴 하는데 한 번 참고해주면 좋을 것 같아.

https://kotlinlang.org/docs/coding-conventions.html#conditional-statements
https://kotlinlang.org/docs/coding-conventions.html#control-flow-statements

Reply:
1. 일단 java를 보완하고자 나온 언어인건 맞는거같은데, 자바의 컨벤션이 그렇다고 해서 그걸 따라야한다는건 좀 아닌거같아. 심지어 코틀린을 위한 컨벤션도 따로 존재하는데 "자바의 컨벤션이 그랬고, 코틀린은 비슷한 언어이니 그걸따르자"는 말이 이상한거같아서 그런 의미에서 명백히 다른 언어라고 말한거야.
2. error-prone라고 말한건, 에러가 절대 안 발생한다고는 말할 수 없다는 의미에서 말한거였어. 위에서 말했듯 IDE 차원에서 인덴트도 다르게 해주는데, 여기서 우리가 인덴트까지 바꿔가면서 코딩을하고 잘못된 결과를 얻는 것은 에러가 발생하지 않을 확률보다 적을테니까

Reply:
3. 당연히 이런 부분은 나도 중요하다고 생각하고, 나 또한 회사에서 코드 리뷰 할 때 작은 부분까지도 많이 이야기를 하기는 해. 만약 팀 내 컨벤션이 정해져있었고 그렇게 정한 타당한 이유가 있었다면 리뷰 할만한 부분이긴 하지.

일단 2번에서 말했듯 이 곳에서 당연히 에러가 발생하지는 않을거라고 생각했어. 그리고, 우리가 컨벤션을 정해놓지도 않았을 뿐더러 자바 컨벤션에서 이랬으니 코틀린에서도 적용하자! 라는 의견은 나한테는 설득력 없게 느껴졌고, 오히려 {}를 하나 더 쓰는게 난 불편해보였어.

만약 "kotlin에서는 if문이 expression이 될 수 있기 때문에 이 경우에 중괄호를 사용하지 않도록 허용해놓은 것이고, 따라서 어떠한 이유가 있으니 사용하지 말자" 였다면 이 논의가 스타일적인 부분이라고 생각하지 않았을거같긴 해 !

이것에 대한 타당성을 찾으려고 스택오버플로우나 여러곳 찾아보다가 android kotlin style guide를 살펴봤는데 여기서는 중괄호를 사용하자는 쪽이네 !

 https://developer.android.com/kotlin/style-guide